cmd/azure.go

```diff
@@ -0,0 +1,515 @@
+	if err != nil {
+		return 0, "", traceError(err)
+	}
+	idStr := string(idByte)
+	splitRes := strings.Split(idStr, ".")
+	if len(splitRes) != 2 {
+		return 0, "", traceError(errUnexpected)
+	}
+	partID, err := strconv.Atoi(splitRes[0])
+	if err != nil {
+		return 0, "", traceError(err)
+	}
+	return partID, splitRes[1], nil
+}
+
+// PutObjectPart - Use Azure equivalent PutBlockWithLength.
+func (a AzureObjects) PutObjectPart(bucket, object, uploadID string, partID int, size int64, data io.Reader, md5Hex string, sha256sum string) (info PartInfo, err error) {
+	id := azureGetBlockID(partID, md5Hex)
+	err = a.client.PutBlockWithLength(bucket, object, id, uint64(size), data, nil)
+	if err != nil {
+		return info, azureToObjectError(traceError(err), bucket, object)
+	}
+	info.PartNumber = partID
+	info.ETag = md5Hex
+	info.LastModified = time.Now().UTC()
+	info.Size = size
+	return info, nil
+}
+
+// ListObjectParts - Use Azure equivalent GetBlockList.
+func (a AzureObjects) ListObjectParts(bucket, object, uploadID string, partNumberMarker int, maxParts int) (result ListPartsInfo, err error) {
+	resp, err := a.client.GetBlockList(bucket, object, storage.BlockListTypeUncommitted)
+	if err != nil {
+		return result, azureToObjectError(traceError(err), bucket, object)
+	}
+	tmpMaxParts := 0
+	partCount := 0 // Used for figuring out IsTruncated.
+	for _, part := range resp.UncommittedBlocks {
+		if tmpMaxParts == maxParts {
+			// Also takes care of the case if maxParts = 0
+			break
+		}
+		partCount++
+		partID, md5Hex, err := azureParseBlockID(part.Name)
+		if err != nil {
+			return result, err
+		}
+		if partID <= partNumberMarker {
+			continue
+		}
+		result.Parts = append(result.Parts, PartInfo{
+			partID,
+			time.Now().UTC(),
+			md5Hex,
+			part.Size,
+		})
+		tmpMaxParts++
+	}
+	result.Bucket = bucket
+	result.Object = object
+	result.UploadID = uploadID
+	result.MaxParts = maxParts
+	if partCount < len(resp.UncommittedBlocks) {
+		result.IsTruncated = true
+	}
+	return result, nil
+}
+
+// AbortMultipartUpload - Not Implemented.
+// There is no corresponding API in azure to abort an incomplete upload. The uncommmitted blocks
+// gets deleted after one week.
+func (a AzureObjects) AbortMultipartUpload(bucket, object, uploadID string) error {
+	return traceError(NotImplemented{})
+}
+
+// CompleteMultipartUpload - Use Azure equivalent PutBlockList.
+func (a AzureObjects) CompleteMultipartUpload(bucket, object, uploadID string, uploadedParts []completePart) (objInfo ObjectInfo, err error) {
+	var blocks []storage.Block
+	for _, part := range uploadedParts {
+		blocks = append(blocks, storage.Block{
+			ID:     azureGetBlockID(part.PartNumber, part.ETag),
+			Status: storage.BlockStatusUncommitted,
+		})
+	}
+	err = a.client.PutBlockList(bucket, object, blocks)
+	if err != nil {
+		return objInfo, azureToObjectError(traceError(err), bucket, object)
+	}
+	return a.GetObjectInfo(bucket, object)
+}
+
+// HealBucket - Not relevant.
```

Comment:
No meaning even in this PR we can move these to a separate file so that they are not written in-between and makes it easy to purge them.  For the next rebase.